xmpp/src/main/java/ch/qos/logback/classic/net/XmppAppender.java

```diff
@@ -0,0 +1,151 @@
+import org.jivesoftware.smack.ChatManager;
+import org.jivesoftware.smack.ConnectionConfiguration;
+import org.jivesoftware.smack.MessageListener;
+import org.jivesoftware.smack.Roster;
+import org.jivesoftware.smack.XMPPConnection;
+import org.jivesoftware.smack.XMPPException;
+import org.jivesoftware.smack.packet.Message;
+
+
+/**
+ * XMMP (Jabber / GoogleHangout) async appender
+ * <p>It is recommended to wrap it with AsyncAppender</p>
+ * @see ch.qos.logback.classic.AsyncAppender
+ * @author szalik
+ */
+public class XmppAppender<E> extends AppenderBase<E> {
+    private ConnectionConfiguration connectionConfiguration;
+    private XMPPConnection conn;
+    private String password;
+    private String username;
+    private String resourceName = getClass().getSimpleName();
+    private String sendToJid;
+    private Chat chat;
+    private int xmmpPort = 5222;
+    private String xmmpServer;
+    private Layout<E> layout = new EchoLayout<E>();
+
+
+    public void setSendToJid(String sendToJid) {
+        this.sendToJid = sendToJid;
+    }
+
+    public void setLayout(Layout<E> layout) {
+        this.layout = layout;
+    }
+
+    /**
+     * @param xmmpAccount jid@server.org[:port]
+     */
+    public void setXmmpAccount(String xmmpAccount) {
+        String[] parts = xmmpAccount.split(":", 2);
+        if (parts.length == 2) {
+            xmmpPort = Integer.parseInt(parts[1].trim());
+        }
+        parts = parts[0].split("@", 2);
+        username = parts[0];
+        xmmpServer = parts[1];
+    }
+
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+
+    public void setPassword(String password) {
+        this.password = password;
+    }
+
+
+    public void setResourceName(String resourceName) {
+        this.resourceName = resourceName;
+    }
+
+
+    @Override
+    public void stop() {
+        boolean doStop = isStarted();
+        super.stop();
+        if (doStop && conn != null && conn.isConnected()) {
+            conn.disconnect();
+            chat = null;
+        }
+    }
+
+    @Override
+    public void start() {
+        Roster.setDefaultSubscriptionMode(Roster.SubscriptionMode.accept_all);
+        connectionConfiguration = new ConnectionConfiguration(xmmpServer, xmmpPort);
+        xmmpConnect();
+        super.start();
+    }
+
+
+    @Override
+    protected void append(E event) {
+        if (chat == null) {
+            addInfo(formatLogMessage("Connecting to xmmp server - " + xmmpServer + ":" + xmmpPort));
+            xmmpConnect();
+        }
+        if (chat != null) {
+            Message message = new Message(sendToJid);
+            message.setBody(layout.doLayout(event));
+            try {
+                chat.sendMessage(message);
+            } catch (XMPPException e) {
+                addError(formatLogMessage("Error sending message to " + sendToJid + "."), e);
+            }
+        }
+    }
+
+    XMPPConnection createXmppConnection() {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/tony19/logback-contrib/file/2373083157/issues/source?bid=3222643&fileBranchId=3222646#l120)